clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -594,7 +594,7 @@ private void resetOffsetIfNeeded(TopicPartition partition, Long requestedResetTi
         } else {
             log.info("Resetting offset for partition {} to offset {}.", partition, offsetData.offset);
             offsetData.leaderEpoch.ifPresent(epoch -> metadata.updateLastSeenEpochIfNewer(partition, epoch));
-            subscriptions.seek(partition, offsetData.offset);
+            subscriptions.maybeSeek(partition, offsetData.offset, requestedResetTimestamp);
```

Comment:
The two checks above were intended to handle this case, but I guess we are missing the synchronization, so do we still need them?

Reply:
Yea, probably we can. Although it might also make sense to move them into `maybeSeek` so we can provide the same logging pattern.

Reply:
Yea I think so. I'll try to find a way to keep those debug loggings though, it might help debugging.